src/TwigComponent/doc/index.rst

```diff
@@ -468,17 +468,32 @@ component use a ``PreMount`` hook::
         {
             // validate data
             $resolver = new OptionsResolver();
+            $resolver->setIgnoreUndefined(true);
+
             $resolver->setDefaults(['type' => 'success']);
             $resolver->setAllowedValues('type', ['success', 'danger']);
             $resolver->setRequired('message');
             $resolver->setAllowedTypes('message', 'string');
 
-            return $resolver->resolve($data);
+            return $resolver->resolve($data) + $data;
         }
 
         // ...
     }
 
+.. note::
+
+   In its default configuration, the OptionsResolver treats all props. 
+   However, if more props are passed than the options defined in the OptionsResolver, an error will be prompted, indicating that one or more options are not defined. 
```

Comment:
@WebMamba `are not defined` is ambiguous.. (not passed or not exist?)

I tend to do:
- `are undefined`
- `do not exist`

```suggestion
   However, if more props are passed than the options defined in the OptionsResolver, an error will be prompted, indicating that one or more options do not exist. 
```

Reply:
I tend more for do not exist